test-common/src/main/java/com/synopsys/integration/alert/test/common/database/MockRepositoryContainer.java

```diff
@@ -0,0 +1,216 @@
+        ids.forEach(dataMap::remove);
+    }
+
+    @Override
+    public void deleteAll(Iterable<? extends T> entities) {
+        entities.forEach(this::delete);
+    }
+
+    @Override
+    public void deleteAll() {
+        dataMap.clear();
+    }
+
+    @Override
+    public <S extends T> @NotNull S save(@NotNull S entity) {
+        ID id = idGenerator.apply(entity);
+        dataMap.put(id, entity);
+        return (S) dataMap.get(id);
+    }
+
+    @Override
+    public <S extends T> @NotNull List<S> saveAll(Iterable<S> entities) {
+        List<S> savedEntities = new LinkedList<>();
+        for (S entity : entities) {
+            savedEntities.add(save(entity));
+        }
+        return savedEntities;
+    }
+
+    @Override
+    public @NotNull Optional<T> findById(@NotNull ID id) {
+        return Optional.ofNullable(dataMap.get(id));
+    }
+
+    @Override
+    public boolean existsById(@NotNull ID id) {
+        return dataMap.containsKey(id);
+    }
+
+    @Override
+    public void flush() {
+        dataMap.clear();
+    }
+
+    @Override
+    public <S extends T> S saveAndFlush(@NotNull S entity) {
+        return save(entity);
+    }
+
+    @Override
+    public <S extends T> List<S> saveAllAndFlush(@NotNull Iterable<S> entities) {
+        return saveAll(entities);
+    }
+
+    @Override
+    public void deleteAllInBatch(@NotNull Iterable<T> entities) {
+        deleteAll(entities);
+    }
+
+    @Override
+    public void deleteAllByIdInBatch(@NotNull Iterable<ID> ids) {
+        deleteAllById(ids);
+    }
+
+    @Override
+    public void deleteAllInBatch() {
+        dataMap.clear();
+    }
+
+    @Override
+    public T getOne(@NotNull ID id) {
+        return dataMap.get(id);
+    }
+
+    @Override
+    public T getById(@NotNull ID id) {
+        return dataMap.get(id);
+    }
+
+    @Override
+    public T getReferenceById(@NotNull ID id) {
+        return dataMap.get(id);
+    }
+
+    @Override
+    public <S extends T> @NotNull Optional<S> findOne(@NotNull Example<S> example) {
+        return Optional.empty();
+    }
+
+    @Override
+    public <S extends T> @NotNull List<S> findAll(@NotNull Example<S> example) {
+        return List.of();
+    }
+
+    @Override
+    public <S extends T> @NotNull List<S> findAll(@NotNull Example<S> example, @NotNull Sort sort) {
+        return List.of();
+    }
+
+    @Override
+    public <S extends T> @NotNull Page<S> findAll(@NotNull Example<S> example, @NotNull Pageable pageable) {
+        return Page.empty();
+    }
+
+    @Override
+    public <S extends T> long count(@NotNull Example<S> example) {
+        return 0;
+    }
+
+    @Override
+    public <S extends T> boolean exists(@NotNull Example<S> example) {
+        return false;
+    }
+
+    @Override
+    public <S extends T, R> R findBy(@NotNull Example<S> example, @NotNull Function<FluentQuery.FetchableFluentQuery<S>, R> queryFunction) {
+        return null;
+    }
```

Comment:
Do we want to implement these methods as well or should we refactor as needed in the future?

Reply:
No we don't want to implement these.  I will add a comment to this section.